lib/Apache/ConfigParser/Directive.pm

```diff
@@ -1071,6 +1116,37 @@ sub directive_value_is_not_dev_null_and_pipe_and_syslog {
   return $_[0] !~ /^\s*(?:(?:\|)|(?:syslog(?::[a-zA-Z0-9]+)?))/;
 }
 
+sub dump {
+    my $node = shift;
+    my $level = shift || 0;
+    my $content = '';
+    my @ds = $node->daughters;
+    if( @ds ) {
+        $content .= '  ' x $level;
+        $content .= '<' . ($node->orig_name||$node->name) . ' ' . ($node->orig_value||$node->value) . ">\n";
+        for my $d ( $node->daughters ) {
+            $content .= $d->dump($level+1);
+        }
+        $content .= '  ' x $level . '</' . ($node->orig_name||$node->name) . ">\n";
+    } else {
+        $content .= '  ' x $level;
+        $content .= ($node->orig_name||$node->name) . ' ' .  ($node->orig_value||$node->value) . "\n";
+    }
+    return $content;
+}
+
+sub find_directive_by_name { 
+    my ($self,$name) = @_;
+    return grep { lc($_->name) eq lc($name) } $self->daughters;
+}
+
+sub remove_directive_by_name {
+    my ($self,$name) = @_;
+    my @nodes = $self->find_directive_by_name($name);
+    return $self->remove_daughter(@nodes);
```

Comment:
Even though http://search.cpan.org/~rsavage/Tree-DAG_Node-1.25/lib/Tree/DAG_Node.pm#remove_daughter(LIST) is a synomyn for remove_daughters(), use remove_daughters() since it's being passed a list.